libraries/tlv-account-resolution/src/state.rs

```diff
@@ -72,6 +72,76 @@ use {
 ///     &remaining_account_infos,
 /// );
 /// ```
+///
+/// Sample usage with PDAs:
+///
+/// ```
+/// use {
+///     solana_program::{
+///         account_info::AccountInfo, instruction::{AccountMeta, Instruction},
+///         pubkey::Pubkey
+///     },
+///     spl_discriminator::{ArrayDiscriminator, SplDiscriminate},
+///     spl_tlv_account_resolution::{
+///         pod::PodAccountMeta,
+///         seeds::Seed,
+///         state::ExtraAccountMetas,
+///    },
+/// };
+///
+/// struct MyInstruction;
+/// impl SplDiscriminate for MyInstruction {
+///     // Give it a unique discriminator, can also be generated using a hash function
+///     const SPL_DISCRIMINATOR: ArrayDiscriminator = ArrayDiscriminator::new([1; ArrayDiscriminator::LENGTH]);
+/// }
+///
+/// // Create a list of required additional accounts for your program.
+/// // This time, you'll have to use `PodAccountMeta`
```

Comment:
To make our lives easier, let's just remove the `init_with_account_metas` bit and force people to always use `PodAccountMeta` -- what do you think? I got a bit confused by the two different ways of doing things

Reply:
We could also rename `PodAccountMeta` to `ExtraAccountMeta` or something, and then make it part of the public API so people know to use that.

We can ditch a lot of the `AccountInfo` helpers, since those were really just done out of convenience. It looks like the interface has already outgrown all of that!